datafusion/sqllogictest/test_files/string_view.slt

```diff
@@ -322,6 +322,378 @@ logical_plan
 03)----TableScan: test projection=[column1_utf8, column2_utf8, column1_utf8view]
 
 
+# Ensure string functions use native StringView implementation
+# and do not fall back to Utf8 or LargeUtf8
+# Should see no casts to Utf8 in the plans below
+
+## Ensure no casts for LIKE/ILIKE
+query TT
+EXPLAIN SELECT
+  column1_utf8view like 'foo' as "like",
+  column1_utf8view ilike 'foo' as "ilike"
+FROM test;
+----
+logical_plan
+01)Projection: test.column1_utf8view LIKE Utf8View("foo") AS like, test.column1_utf8view ILIKE Utf8View("foo") AS ilike
+02)--TableScan: test projection=[column1_utf8view]
+
+
+
+## Ensure no casts for ASCII
+## TODO file ticket
```

Comment:
We plan to file tickets for these functions as follow ons -- see https://github.com/apache/datafusion/pull/11787#discussion_r1702294173 for details

Reply:
Here's the one for ASCII: https://github.com/apache/datafusion/issues/11834

I don't have the ability to update the label, so maybe it could be changed to a good first issue? Might help it get picked up sooner, and with the examples, it's not doing it from scratch.